napari/_qt/qt_main_window.py

```diff
@@ -1730,6 +1731,67 @@ def export_figure(
             imsave(path, img)
         return img
 
+    def export_rois(
+        self,
+        rois: np.ndarray,
+        paths: list[str] | None = None,
+        scale: float | None = None,
+    ):
+        """Export the shapes rois with storage file paths.
+
+        Parameters
+        ----------
+        rois: np.ndarray
+            An array of shape (n, 2, 2) where n is the number of rois
+            and the first two coordinates correspond to the top left and
+            the last two coordinates correspond to the bottom right corners
+        paths: list
+            The list to store file path for shapes roi
+
+        Returns
+        -------
+        screenshot_list: list
+            The list with roi screenshots.
+
+        """
+        if paths is not None and len(paths) != len(rois):
+            raise ValueError(
+                trans._(
+                    'The number of file paths does not match the number of ROI shapes',
+                    deferred=True,
+                )
+            )
+
+        screenshot_list = []
+        camera = self._qt_viewer.viewer.camera
+        start_camera_center = camera.center
+        start_camera_zoom = camera.zoom
+        canvas = self._qt_viewer.canvas
+        prev_size = canvas.size
+
+        for index, shape in enumerate(rois):
+            top_left = shape[0]
+            bottom_right = shape[1]
+            x1, y1 = top_left
+            x2, y2 = bottom_right
+            if x1 > x2 or y1 > y2:
+                raise ValueError('ROI shape error')
+            center_coord, height, width = get_center_bbox(shape)
+            camera.center = center_coord
+            canvas.size = (int(height), int(width))
```

Comment:
Since `center` is floats and `canvas.size` are integers, I think we may have a problem with centering correctly on pixels. Did you run some tests manually to ensure that this does what you expect?

Reply:
Thanks for catching this @brisvag we tested this manually and it seemed to work ok. I agree that this could be an issue down the road, what do you think about rounding up the floats to nearest integer as a resolution?

Reply:
Thanks @brisvag and that makes sense. I will add a comment regarding potential alignment issues as suggested.